src/sha2.rs

```diff
@@ -0,0 +1,61 @@
+use crate::HASH_LEN;
+use sha2::{Digest, Sha256};
+
+/// Returns the digest of `input`.
+pub fn hash(input: &[u8]) -> Vec<u8> {
+    let mut hasher = Sha256::new();
+    hasher.update(input);
+    hasher.finalize().to_vec()
+}
+
+/// Hash function returning a fixed-size array (to save on allocations).
+pub fn hash_fixed(input: &[u8]) -> [u8; HASH_LEN] {
+    let mut hasher = Sha256::new();
+    hasher.update(input);
+    hasher.finalize().into()
+}
+
+/// Compute the hash of two slices concatenated.
+pub fn hash32_concat(h1: &[u8], h2: &[u8]) -> [u8; 32] {
+    let mut hasher = Sha256::new();
+    hasher.update(h1);
+    hasher.update(h2);
+    hasher.finalize().into()
+}
+
+/// Context for incremental hashing
+pub struct Context(Sha256);
+
+impl Context {
+    pub fn new() -> Self {
+        Self(Sha256::new())
+    }
+
+    pub fn update(&mut self, bytes: &[u8]) {
+        self.0.update(bytes);
+    }
+
+    pub fn finalize(self) -> [u8; HASH_LEN] {
+        self.0.finalize().into()
+    }
+}
+
```

Comment:
```suggestion

#[cfg(feature = "sha2")]
```

Test for `sha2` runs even in `feature = "poseidon2"`.